python/lsst/daf/butler/core/dimensions/dataId.py

```diff
@@ -511,6 +512,26 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(frozenset(self._linkValues.items()))
 
+    def updateHash(self, message):
+        """Add this data ID to a secure hash.
+
+        Parameters
+        ----------
+        `message` : `hashlib` message instance
+            Object with an ``update`` method that takes a single `bytes`
+            argument to update the hash.
+        """
+        for k, v in self.items():
+            message.update(k.encode('utf8'))
+            if isinstance(v, int):
+                message.update(v.to_bytes(64, byteorder='little'))
+            elif isinstance(v, str):
+                message.update(v.encode('utf8'))
+            elif isinstance(v, datetime.datetime):
+                message.update(v.isoformat().encode('utf8'))
```

Comment:
I *think* it does, but I'm not certain, and I didn't see an easy way to get number of seconds in UTC from Python `datetime`.  In any case, I expect this all to get fixed in [DM-15890|https://jira.lsstcorp.org/browse/DM-15890], which will switch all of the datetimes we use in the database to TAI MJD to avoid these kinds of problems entirely.